python/pfs/drp/stella/estimateRadialVelocity.py

```diff
@@ -0,0 +1,178 @@
+        error : `float`
+            Standard deviation of ``velocity``.
+            This is reliable only if ``config.findMethod="gauss"``
+            and ``config.useCovar=True``.
+        """
+        # TODO: This method should be wholly rewritten so that it will use
+        # a log-scaled wavelength for the sake of FFT convolution.
+
+        searchMin, searchMax, searchStep = self.config.searchRange
+        searchNum = 1 + int(math.ceil((searchMax - searchMin) / searchStep))
+        searchVelocity = np.linspace(searchMin, searchMax, num=searchNum, endpoint=True)
+        beta = searchVelocity / const.c.to('km/s').value
+        doppler = np.sqrt((1.0 + beta) / (1.0 - beta))
+
+        wavelength = spectrum.wavelength
+        flux = spectrum.flux - 1.0
+        covar = spectrum.covar
+
+        modelWavelength = modelSpectrum.wavelength
+        modelFlux = modelSpectrum.flux - 1.0
+
+        # Note: scaledModel[i] = modelSpectrum moving at searchVelocity[i]
+        scaledModel = np.empty(shape=(len(searchVelocity), len(wavelength)))
+        for i, dopp in enumerate(doppler):
+            # We divide the model flux by `dopp`
+            # assuming that line spectra contribute to the correlation
+            # much more than the continuum, not subtracted perfectly, does.
+            # If this assumption is wrong, we must not divide it by `dopp`.
+            scaledModel[i, :] = scipy.interpolate.interp1d(
+                modelWavelength * dopp,
+                modelFlux,
+                kind="linear",
+                bounds_error=False,
+                fill_value=0
+            )(wavelength) / dopp
+
+        # This is cross correlation function
+        ccf = scaledModel @ flux
+
+        # Find the peak of CCF
+        if self.config.findMethod == "peak":
+            iMax = np.argmax(ccf)
+            velocity = searchVelocity[iMax]
+            # We have to think of the error...
+            return velocity, 0
+
+        # Gaussian fit
+        if self.config.findMethod == "gauss":
+            def gauss(v, a, v_est, sigma):
+                return a * np.exp((v - v_est)**2 / (-2*sigma**2))
+
+            iMax = np.argmax(ccf)
+            velocity = searchVelocity[iMax]
+            coeff = 1.0 / ccf[iMax]
+            fitIndex = (searchVelocity > (velocity - self.config.peakRange / 2)) & \
+                       (searchVelocity < (velocity + self.config.peakRange / 2))
+            fitVelocity = searchVelocity[fitIndex]
+            fitCcf = coeff * ccf[fitIndex]
+            fitCovar = transformCovar((coeff * coeff) * covar, scaledModel[fitIndex, :])
+            if not self.config.useCovar:
+                # `scipy.optimize.curve_fit()` takes standard deviation
+                # rather than variance if `sigma` argument is not a matrix.
+                fitCovar = np.sqrt(np.diag(fitCovar))
+
+            iniParam = [1.0, velocity, self.config.peakRange]
+            pfit, pcov = scipy.optimize.curve_fit(
```

Comment:
I checked it, but the variance (covariance) argument is essential here. Yamashita-san says he will use, if available, the velocity error (`sqrt(pcov[1])`) somewhere in the latter part of the analysis. (To make the velocity error correct, variance is not sufficient. Covariance matters very much here because the sample points fit to are cross-correlation)